### ISSUE TITLE
Verilog syntax highlighting/filetype association broken in `filetype.vim`?

### ISSUE THREAD

**Issue by ZakSN** [vim/vim#15694]
I'm using Arch Linux's `vim-runtime 9.1.0707-1` package and noticed that Verilog syntax highlighting stopped working. I found that editing this line:

https://github.com/vim/vim/blob/f21d28a5c72987d42d540290e5a77500ab1a4d7c/runtime/filetype.vim#L2623

to:

`au BufNewFile,BufRead *.v			setf verilog`

seems to fix the problem. I'm not sure if that's a correct solution, maybe someone more knowledgeable than me can comment...

**Comment by chrisbra** [comment]
Does it happen for all verilog files and is it perhaps detected as a different filetype? Can you find out, why [ft.vim](https://github.com/vim/vim/blob/f21d28a5c72987d42d540290e5a77500ab1a4d7c/runtime/autoload/dist/ft.vim#L1361-L1412) does not detect your filetype?

**Comment by ZakSN** [comment]
It actually seems like it is only missing some files. I observed this behaviour while working with the Corundum project (https://github.com/corundum/corundum/tree/master), and it seems like syntax highlighting is broken on at least these files:
```
corundum/fpga/common/rtl/mqnic_core_axi.v
corundum/fpga/common/rtl/mqnic_core_pcie_ptile.v
corundum/fpga/common/rtl/mqnic_core_pcie_s10.v
corundum/fpga/common/rtl/mqnic_core_pcie_us.v
corundum/fpga/common/rtl/mqnic_core_pcie.v
corundum/fpga/common/rtl/mqnic_core.v
```

but works for other files in that directory e.g.:
```
corundum/fpga/common/rtl/rb_drp.v
corundum/fpga/common/rtl/mqnic_ptp.v
```

if I run `set filetype?` in `mqnic_core.v` I see: `filetype=v`, whereas if I run `set filetype?` in `rb_drp.v` I see: `filetype=verilog`

I'm not very familiar with vim script, but I'll try taking a look at ft.vim.

**Comment by ZakSN** [comment]
A quick guess: the files that the syntax highlighting fails on seem to all open with very large module declarations e.g. from:
https://github.com/corundum/corundum/blob/master/fpga/common/rtl/mqnic_core.v#L15
to
https://github.com/corundum/corundum/blob/master/fpga/common/rtl/mqnic_core.v#L200
is all a module declaration, none of which end with `;`, so I think ft.vim doesn't interpret it as verilog.

**Comment by chrisbra** [comment]
Yep we only check the very first 200 lines. It seems the `;` only starts at line 500, but we would need to reload quite a bit for lines then. 
How about you instead define inside your .vimrc:
```
let g:filetype_v='verilog'
```
So Vim does not have to inspect the files at all?

**Comment by ZakSN** [comment]
adding `let g:filetype_v='verilog' to my `.vimrc` seems to turn on highlighting again.
